### DISCUSSION TITLE
How to run 'Create file image' and/or 'Print size' tools?

### DISCUSSION THREAD

**Discussion by SteMMo33** [eclipse-embed-cdt/eclipse-plugins#616] [General Discussions]
Hi all,
I defined a embedded ARM Cortex-A7 project, adding compiler and linker options.
How can I add the run of 'GNU Arm Cross create flash Image' and/or 'GNU Arm Cross Prin size' tools in the compile flow?

Thanks!

**Comment by TommyMurphyTM1234** [comment]
The plugins create projects that have builds steps that run those tools by default as far as I recall. 

- https://eclipse-embed-cdt.github.io/plugins/features/#extra-build-steps
- https://eclipse-embed-cdt.github.io/tutorials/blinky-arm/#build-the-project

You'll probably need to explain in more detail how exactly you created your project and what you ended up with because it doesn't sound like you used the plugins as normal/recommended.

**Comment by SteMMo33** [comment]
Oh, yes, you are right!
The very first projects was done by the scratch, and they have these steps.
The one I refer is imported and it is originally wrote for STMCubeIDE.
Is there a way to add these steps?

Thanks!

**Reply by TommyMurphyTM1234** [reply]
> The one I refer is imported and it is originally wrote for STMCubeIDE. Is there a way to add these steps?

Isn't that really a question for the STMCubeIDE community? 
If you want to use Embedded CDT to manage/build your code then you need to create an Embedded CDT project for it. 
If you want to use the STM approach and want to add extra steps then that's outside the scope of this repo.
